recipes/kokkos-kernels/meta.yaml

```diff
@@ -0,0 +1,60 @@
+{% set name = "kokkos-kernels" %}
+{% set version = "3.7.01" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/kokkos/kokkos-kernels/archive/refs/tags/{{ version }}.tar.gz
+  sha256: b2060f5894bdaf7f7d4793b90444fac260460cfa80595afcbcb955518864b446
+
+build:
+  skip: true  # [win]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('kokkos-kernels') }}
```

Comment:
I'm not 100% sure, but I would say yes.

Reply:
Hi Leo, I think I misunderstood you question here. I had in mind to match the versions of Kokkos and Kokkos-kernels, but I realize this has nothing to do with it. So I guess we might remove that, but I'm also wondering what's the best way to lock these together. Just 
```
  host:
    - kokkos =={{ version }}
```
?

Reply:
@vincentmr I see. My question was "for those who depend on Kokkos Kernels, would they pin kokkos-kernels tightly that they use the same version at run time as at build time?" which is what this `run_exports` is about.

Your concern was that "Kokkos Kernels depends on Kokkos, will they be version-matched?" The answer is currently no. The [`run_exports` of Kokkos](https://github.com/conda-forge/kokkos-feedstock/blob/ba2a1102b1b9e99adc5afb1a62e4b93789822d81/recipe/meta.yaml#L19-L20) only guarantees that the run time and build time `kokkos` is the same version, but it's not necessary the same as the `kokkos-kernels` version. We should fix this in the feedstock.

Reply:
In particular, note that kokkos v4.0.0 just came out. It seems `kokkos-kernels` v3.7.01 should be rebuilt, and the build number 0 should be marked as broken (since they depend on `kokkos` v4.0.0).

Reply:
> It seems `kokkos-kernels` v3.7.01 should be rebuilt, 

Ah, I just saw that you already did it: https://github.com/conda-forge/kokkos-kernels-feedstock/pull/1, thanks!

> and the build number 0 should be marked as broken (since they depend on `kokkos` v4.0.0).

This needs to be done.

Reply:
> In particular, note that kokkos v4.0.0 just came out a few weeks ago. It seems `kokkos-kernels` v3.7.01 should be rebuilt, and the build number 0 should be marked as broken (since they depend on `kokkos` v4.0.0).

I thought iterating the build number would forcefully deprecate lower build numbers. Ok, so let me try to do that. And thx for the feedback.

Reply:
> I thought iterating the build number would forcefully deprecate lower build numbers.

Usually it's the case. But I've seen some complicated situations where the solver thinks it's better to pick the ones with a lower build number. Plus, in this case it's literally broken; we're not supposed to use Kokkos 4 to build Kokkos Kernels 3, unless there's strong compatibility guarantee from the Kokkos team.